source/opt/fold_spec_const_op_composite.cpp

```diff
@@ -252,10 +251,10 @@ std::vector<uint32_t> OperateVectors(
 bool FoldSpecConstantOpAndCompositePass::ProcessImpl(ir::Module* module) {
   bool modified = false;
   auto inst_iter = module->types_values_begin();
-  // In case the types, constants, global variabls secion is empty, do nothing.
-  if (inst_iter == module->types_values_end()) return modified;
-
-  do {
+  for (;
```

Comment:
You could put the auto inst_iter = .... right into the for( construct.  It usually goes there.
And you can pull the comment out to just before.  That way it looks completely idiomatic.